code-scanning/roxctl.yml

```diff
@@ -0,0 +1,57 @@
+# This workflow uses actions that are not certified by GitHub.
+# They are provided by a third-party and are governed by
+# separate terms of service, privacy policy, and support
+# documentation.
+
+# This workflow integrates roxctl with GitHub's Code Scanning
+# feature.
+#
+# roxctl is the CLI of StackRox, a container security platform.
+# roxctl will scan container images for vulnerabilities.
+#
+# See https://www.stackrox.io/docs/ for more on StackRox.
+#
+# See https://github.com/stackrox/roxctl-installer-action for
+# more on the roxctl-installer action and its parameters.
+
+name: Roxctl
+
+on:
+  push:
+    branches: [ $default-branch, $protected-branches ]
+  pull_request:
+    branches: [ $default-branch ]
+  schedule:
+    - cron: $cron-weekly
+
+permissions:
+  contents: read
+
+jobs:
+  roxctl-image-scan:
+    permissions:
+      # Only required for private repositories by github/codeql-action/upload-sarif
+      # to get the Action run status.
+      actions: read
+      # Required for actions/checkout to fetch code.
+      contents: read
```

Comment:
This is only required for private repos right? Since we call that out in the line above, we may as well here.

Reply:
No, it's always required, it's just that _by default_ it is always set to `read` (see [this doc](https://docs.github.com/en/actions/security-guides/automatic-token-authentication#permissions-for-the-github_token)). I just followed the pattern from a couple of other actions which explicitly documented this here in case folks mess around with their GITHUB_TOKEN config.